src/modm/math/filter/moving_average_float_impl.hpp

```diff
@@ -53,31 +50,30 @@ namespace modm
 //---------------------------------------------------------------------------------
 
 template<std::size_t N>
-modm::filter::MovingAverage<double, N>::MovingAverage(const double& initialValue) :
-	index(0), sum(N * initialValue)
+modm::filter::MovingAverage<double, N>::MovingAverage(const double& initialValue)
 {
-	for (Index i = 0; i < N; ++i) {
-		buffer[i] = initialValue;
-	}
+	reset(initialValue);
 }
 
+// ----------------------------------------------------------------------------
+template<std::size_t N>
+void
+modm::filter::MovingAverage<double, N>::reset(const double& input)
+{
+	std::fill(std::begin(buffer), std::end(buffer), input);
+	sum = N * input;
+}
 
 //---------------------------------------------------------------------------------
 template<std::size_t N>
 void
 modm::filter::MovingAverage<double, N>::update(const double& input){
 	buffer[index] = input;
 
-	index++;
-	if (index >= N) {
+	if (++index >= N)
 		index = 0;
-	}
-
-	sum = 0;
-	for (Index i = 0; i < N; ++i) {
-		sum+= buffer[i];
-	}
 
+	sum = std::accumulate(std::begin(buffer), std::end(buffer), 0);
 }
```

Comment:
If I see this correctly, the only difference is the sum computation, but can't this be specialized inside the function with a `if constexpr`? That would save a lot of duplicated code.

Reply:
Yea come on, then let's also inline the method definitions and finish this up!